### PR TITLE
Cleanup run_in_background configuration and use ENV instead

### PR DESCRIPTION
**Pull request by abenbachir**

### CODE REVIEW

File: source/code/plugins/out_oms.rb [hunk 1]
```diff
@@ -29,6 +29,11 @@ def initialize
     def configure(conf)
       s = conf.add_element("secondary")
       s["type"] = ChunkErrorHandler::SecondaryName
+
+      # override default run_in_background config
+      if !OMS::BackgroundJobs::enabled.nil?
+        conf['run_in_background'] = OMS::BackgroundJobs::enabled
```
Comment (hestolz) [inline source/code/plugins/out_oms.rb]: you can reassign conf dict? why not just reassign `@run_in_background`

**Comment by abenbachir** [comment]
> How many customers are currently relying on this?
> How are you going to avoid breaking this feature for them?

And no one is using it for now. 
Nothing will break, I'm introducing a new ENV variable o handle turning on/off that feature.

File: source/code/plugins/out_oms.rb (hunk 1 above)
Comment (abenbachir) [reply source/code/plugins/out_oms.rb]: If you assign @run_in_background fluentd will reset that value from "conf['run_in_background']".
Look at other fluentd internal plugins, they all use this technique to override configs.

Comment (hestolz) [reply source/code/plugins/out_oms.rb]: yeah after i commented i checked, whoops

**Pull request merged**